### ISSUE TITLE
Namespace scheduler requeues a namespace infinitely if no clusters are registered

### ISSUE THREAD

**Issue by ncdc** [kcp-dev/kcp#433]
1. Start kcp with --install-namespace-scheduler
2. Create a namespace
3. See logs like this:

```
I0204 09:53:21.309079   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:21.309105   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:21.314534   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:21.314567   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:21.324688   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:21.324716   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:21.344862   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:21.344899   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:21.386025   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:21.386061   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:21.466640   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:21.466673   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:21.626991   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:21.627024   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:21.947854   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:21.947919   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:22.588477   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:22.588514   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:23.868711   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:23.868751   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:26.428935   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:26.428965   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
I0204 09:53:31.549336   66876 namespace.go:109] Reconciling Namespace default
E0204 09:53:31.549365   66876 controller.go:308] "namespace-scheduler" controller failed to sync "admin#$#default", err: no clusters
```

This is because https://github.com/kcp-dev/kcp/blob/2f149e7d72bec5e2cfca671a25167de9eb417cff/pkg/reconciler/namespace/namespace.go#L129 returns an error, which bubbles up to `processNextNamespace`, which then does a rate-limited requeue.

I'm not sure this is a desirable behavior?

**Comment by marun** [comment]
Will be fixed by https://github.com/kcp-dev/kcp/pull/428

**Comment by marun** [comment]
Fixed by #428
